src/ape/contracts/base.py

```diff
@@ -463,6 +463,19 @@ def __len__(self):
         logs = self.provider.get_contract_logs(self.log_filter)
         return sum(1 for _ in logs)
 
+
+    def __call__(self, *args: Any, **kwargs: Any) -> "BaseContractLog":
+        event_args = dict(zip([input.name for input in self.abi.inputs], args))
+        event_args.update(kwargs)
```

Comment:
Maybe add a disjoint set between `event_args.keys()` and `kwargs.keys()` to prevent overwrites

Reply:
Also add a check after updating that `event_args.keys()` is a subset of `set(input.name for input in self.abi.inputs)`